example5.ts

```diff
@@ -0,0 +1,80 @@
+import * as readline from 'readline';
+
+type Position = {
+  x: number;
+  y: number;
+};
+
+const maze: string[][] = [
+  [' ', ' ', ' ', '*', ' ', ' ', ' '],
+  ['*', '*', ' ', '*', ' ', '*', ' '],
+  [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+  [' ', '*', '*', '*', '*', '*', ' '],
+  [' ', ' ', ' ', ' ', ' ', ' ', 'E'], // Exit
+];
+
+let playerPosition: Position = { x: 0, y: 0 }; // Starting position
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+function printMaze(): void {
+  maze.forEach((row, y) => {
+    let rowStr: string = '';
+    row.forEach((cell, x) => {
+      if (playerPosition.x === x && playerPosition.y === y) {
+        rowStr += 'P ';
+      } else {
+        rowStr += cell + ' ';
+      }
+    });
+    console.log(rowStr);
+  });
+}
+
+function movePlayer(direction: string): boolean {
+  let newX = playerPosition.x;
+  let newY = playerPosition.y;
+
+  switch (direction) {
+    case 'w': newY -= 1; break;
+    case 's': newY += 1; break;
+    case 'a': newX -= 1; break;
+    case 'd': newX += 1; break;
+    default: console.log('Invalid direction! Use w, a, s, d to move.'); return false;
```

Comment:
Consider throwing an error instead of logging a message for an invalid direction input. This could help users understand the issue better. Your overconfidence is your weakness.